src/charm.py

```diff
@@ -213,74 +146,70 @@ def get_manifest(self):
-            "config_hash": config_hash.hexdigest(),
-        }
+        # Get current layer
+        current_layer = self._container.get_plan()
+        # Create a new config layer
+        new_layer = self._dex_auth_layer
+        if current_layer.services != new_layer.services:
+            self.unit.status = MaintenanceStatus("Applying new pebble layer")
+            self._container.add_layer(self._container_name, new_layer, combine=True)
+            self.logger.info("Pebble plan updated with new configuration")
 
-        return [
-            obj
-            for path in glob("src/manifests/*.yaml")
-            for obj in codecs.load_all_yaml(Path(path).read_text(), context=context)
-        ]
+        # Get current dex config
+        current_config = self._container.pull(self._dex_config_path).read()
+        if current_config != config:
+            self._container.push(self._dex_config_path, config, make_dirs=True)
+            self.logger.info("Updated dex config")
 
-    def _check_deployed_resources(self, manifest=None):
-        """Check the status of deployed resources, returning True if ok else raising CheckFailedError
+        # Using restart due to https://github.com/canonical/dex-auth-operator/issues/63
+        self._container.restart(self._container_name)
 
-        All abnormalities are captured in logs
+    def main(self, event):
+        try:
+            self._check_leader()
+        except CheckFailedError as err:
+            self.model.unit.status = err.status
+            return
 
-        Params:
-          manifest: (Optional) list of lightkube objects describing the entire application.  If
-                    omitted, will be computed using self.get_manifest()
-        """
-        if manifest:
-            expected_resources = manifest
-        else:
-            expected_resources = self.get_manifest()
-        found_resources = [None] * len(expected_resources)
-        errors = []
+        self.model.unit.status = MaintenanceStatus("Configuring dex charm")
+        self.ensure_state()
 
-        client = Client()
+        self._update_layer()
 
-        self.logger.info("Checking for expected resources")
-        for i, resource in enumerate(expected_resources):
-            try:
-                found_resources[i] = client.get(
-                    type(resource),
-                    resource.metadata.name,
-                    namespace=resource.metadata.namespace,
-                )
-            except ApiError:
-                errors.append(
-                    f"Cannot find k8s object for metadata '{resource.metadata}'"
-                )
-
-        self.logger.info("Checking readiness of found StatefulSets/Deployments")
-        statefulsets_ok, statefulsets_errors = validate_statefulsets_and_deployments(
-            found_resources
+        try:
+            self.handle_ingress()
+        except Exception as err:
+            self.model.unit.status = BlockedStatus(str(err))
+            return
```

Comment:
Should we just let these `Exception`s raise to Juju?  It feels odd to catch **all** exceptions here and set them to Blocked.  I think the usual practice would be to set Waiting/Blocked when we know what happened, but to let exceptions raise (and the program error out) if we don't know (resulting in an `ErrorStatus` from Juju)